test/delete_test.go

```diff
@@ -88,3 +91,67 @@ func TestSkiplineComputationAfterUpgrade(t *testing.T) {
 	assert.Equal(t, "00000001000000000000009C", walSkipFileName)
 	assert.Equal(t, 3, skipLine)
 }
+
+func TestDeleteOldObjects(t *testing.T) {
+	folder := createMockStorageFolder()
+	expectedOnlyOneSavedObjectName := "basebackups_005/base_123312"
+	filter := func(object storage.Object) bool {
+		return object.GetName() != expectedOnlyOneSavedObjectName
+	}
+	err := internal.DeleteOldObjects(folder, filter)
+	assert.NoError(t, err)
+	savedObjects, err := storage.ListFolderRecursively(folder)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(savedObjects))
+	assert.Equal(t, expectedOnlyOneSavedObjectName, savedObjects[0].GetName())
+}
+
+func TestFindTargetBeforeName_ReturnsErrorForDeltaBackup_Without_Modifier(t *testing.T) {
+	targetDelta := "base_000000010000000000000009_D_000000010000000000000007"
+	folder := createMockStorageFolderWithDeltaBackups(t)
+	expectedErrorMessage := " is incremental and it's predecessors cannot be deleted. Consider FIND_FULL option."
```

Comment:
You didn't check the error message in this code. Anyway, it's better to check the error type or to check only error existence.

Reply:
Why? If you don't check the message, you can't be sure which case you test.

Reply:
Then, let's do a separate type for it and check the error type. It is better because of the flexibility of this variant (If you change error message than you shouldn't change test. Also, the test shouldn't test the exact  error message, it should test logic behind the error).

Reply:
done